From 8129dc7af9fc378c94ef0da9f8d2eca5d8385983 Mon Sep 17 00:00:00 2001
From: Hans Petter Selasky <hps@selasky.org>
Date: Sat, 8 Oct 2016 17:37:17 +0200
Subject: [PATCH 1052/1936] Build fixes for FreeBSD.

Check for dlopen in Lib-C and make sure the lv2-ttl-generator is
linked with -lpthread, else loading some shared objects with X11
support might fail under FreeBSD.

Signed-off-by: Hans Petter Selasky <hps@selasky.org>
---
 src/Plugin/CMakeLists.txt | 13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/src/Plugin/CMakeLists.txt b/src/Plugin/CMakeLists.txt
index 98f37d93..06c4bcc7 100644
--- a/src/Plugin/CMakeLists.txt
+++ b/src/Plugin/CMakeLists.txt
@@ -5,6 +5,19 @@ CHECK_LIBRARY_EXISTS(dl dlopen "" LIBDL_FOUND)
 
 IF(LIBDL_FOUND)
 target_link_libraries(lv2-ttl-generator "dl")
+ELSE(LIBDL_FOUND)
+# FreeBSD has dlopen as part of LibC
+CHECK_LIBRARY_EXISTS(c dlopen "" LIBDL_FOUND)
+ENDIF(LIBDL_FOUND)
+
+CHECK_LIBRARY_EXISTS(pthread pthread_create "" LIBPTHREAD_FOUND)
+
+IF(LIBPTHREAD_FOUND)
+# Some X11 libGL libraries require -lpthread on FreeBSD
+target_link_libraries(lv2-ttl-generator "pthread")
+ENDIF(LIBPTHREAD_FOUND)
+
+IF(LIBDL_FOUND)
 add_subdirectory(AlienWah)
 add_subdirectory(Chorus)
 add_subdirectory(Distortion)
-- 
2.47.0

